rosdoc2/verbs/build/builders/sphinx_builder.py

```diff
@@ -372,6 +388,48 @@ def build(self, *, doc_build_folder, output_staging_directory):
             sourcedir,
             intersphinx_mapping_extensions)
 
+        # If the package has build type `ament_python`, or if the user configured
+        # to run `sphinx-apidoc`, then invoke `sphinx-apidoc` before building
+        if (self.build_context.build_type == 'ament_python' or
+                self.build_context.always_run_sphinx_apidoc):
+            package_xml_directory = os.path.dirname(self.build_context.package.filename)
+            # If 'python_source' is specified, construct 'package_src_directory' from it
+            if self.build_context.python_source is not None:
+                package_src_directory = \
+                    os.path.join(
+                        package_xml_directory,
+                        self.build_context.python_source)
+            # If not provided, try to find the package source direcotry
+            else:
+                package_list = setuptools.find_packages(where=package_xml_directory)
```

Comment:
I'll be interested to see what happens here when we scale this out to more of our python packages. In the past when using `setuptools.find_packages()` in my `setup.py` files I've had to exclude test directories and stuff like that.

Reply:
Yes, this is the first thing I've noticed too. One immediate idea is to do a wildcard search such as `[pkg-name]*`, and only add those modules to the path.